### PR TITLE
Fix showname placeholder text and shout button after changing characters

### PR DESCRIPTION
**Pull request by in1tiate**
- Changing character indirectly (such as through server commands or other methods that skip the character select screen) will now update the showname box's placeholder text to your new character's name.
- Fix for #116.

### CODE REVIEW

**Comment by Crystalwarrior** [comment]
Since 2.7 is skipped over, would it be possible to change the base branch to kfo-ao2?

**Comment by in1tiate** [comment]
@Crystalwarrior Rebased to `kfo-ao2`, upon initial testing seems to work just fine.

**Comment by Crystalwarrior** [comment]
Huh why did the checks fail

File: src/courtroom.cpp [hunk 1]
```diff
@@ -1262,6 +1262,9 @@ void Courtroom::update_character(int p_cid)
   else
     ui_custom_objection->hide();
 
+  if (m_cid != -1) // there is no name at char_list -1, and we crash if we try
+                   // to find one
+    ui_ic_chat_name->setPlaceholderText(char_list.at(char_id).name);
```
Comment (Crystalwarrior) [inline src/courtroom.cpp:1267]: oh just rename char_id to m_cid, seems like a simple fix

**Comment by oldmud0** [comment]
wait did i just approve failing code

**Pull request merged**